python/feldera/_helpers.py

```diff
@@ -15,14 +56,37 @@ def ensure_dataframe_has_columns(df: pd.DataFrame):
         )
 
 
-def dataframe_from_response(buffer: list[list[dict]]):
+def dataframe_from_response(buffer: list[list[dict]], schema: dict):
     """
     Converts the response from Feldera to a pandas DataFrame.
     """
-    return pd.DataFrame([
+
+    pd_schema = {}
+
+    decimal_col = []
+
+    for column in schema['fields']:
+        column_name = column['name']
+        column_type = column['columntype']['type']
+        if column_type == 'DECIMAL':
+            decimal_col.append(column_name)
+
+        pd_schema[column_name] = sql_type_to_pandas_type(column_type)
+
+    data = [
         {**item['insert'], 'insert_delete': 1} if 'insert' in item else {**item['delete'], 'insert_delete': -1}
         for sublist in buffer for item in sublist
-    ])
+    ]
+
+    for datum in data:
+        for col in decimal_col:
+            if datum[col] is not None:
+                datum[col] = Decimal(datum[col])
+
+    df = pd.DataFrame(data)
```

Comment:
so dataframe sets types with a mix of type annotations and extracting the decimal type from values?
this sure is an ugly API

Reply:
I didn't quite find a way to good way to define the table and types before creating a dataframe.
 
So, some get converted with `astype` to the schema, and for decimal, as this is not supported, we create the object before passing it to the dataframe. 

This is pretty ugly, yeah.